hikari/impl/special_endpoints.py

```diff
@@ -1150,48 +1150,21 @@ def set_user_mentions(
 
     def build(
         self, entity_factory: entity_factory_.EntityFactory, /
-    ) -> typing.Tuple[typing.MutableMapping[str, typing.Any], typing.Sequence[files.Resource[files.AsyncReader]]]:
-        data = data_binding.JSONObjectBuilder()
-        data.put("content", self.content)
-
-        final_attachments = []
-        if self._attachments:
-            attachments_payload = []
-
-            for f in self._attachments:
-                if isinstance(f, messages.Attachment):
-                    attachments_payload.append({"id": f.id, "filename": f.filename})
-                    continue
-
-                final_attachments.append(files.ensure_resource(f))
-
-            if attachments_payload:
-                data.put("attachments", attachments_payload)
-
-        elif self._attachments is None:
-            data.put("attachments", None)
-
-        if self._embeds is not undefined.UNDEFINED:
-            embeds: typing.List[data_binding.JSONObject] = []
-            for embed, attachments in map(entity_factory.serialize_embed, self._embeds):
-                final_attachments.extend(attachments)
-                embeds.append(embed)
-
-            data["embeds"] = embeds
-
-        data.put_array("components", self._components, conversion=lambda component: component.build())
-        data.put("flags", self.flags)
-        data.put("tts", self.is_tts)
-
-        if (
-            not undefined.all_undefined(self.mentions_everyone, self.user_mentions, self.role_mentions)
-            or self.type is base_interactions.ResponseType.MESSAGE_CREATE
-        ):
-            data["allowed_mentions"] = mentions.generate_allowed_mentions(
-                self.mentions_everyone, undefined.UNDEFINED, self.user_mentions, self.role_mentions
-            )
+    ) -> typing.Tuple[typing.MutableMapping[str, typing.Any], typing.Mapping[int, files.Resource[files.AsyncReader]]]:
+        data, to_upload = message_utils.build_message_payload(
```

Comment:
Adding support for the edit semantics around passing `None` for fields and allowing using `messages.Attachment` to avoid re-uploading attachments should be as easy as deciding whether `edit` should be True or False based on `self._type` and allowing None to be passed for some of these fields (I just haven't done that yet since that wasn't the focus of this PR)